[Merged by Bors] - feat(Algebra/Order): add Finite(Mul)ArchimedeanClass

wwylele opened the pull request:
Separated from #25970. This adds a subtype of (Mul)ArchimedeanClass that is useful in formalizing Hahn embedding theorem.

The only code change since last review was doc string update for `to_additive`

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include at least one commit authored by each
co-author among the commits in the pull request. If necessary, you may 
create empty commits to indicate co-authorship, using commands like so:

git commit --author="Author Name <author@email.com>" --allow-empty -m "add Author Name as coauthor"

When merging, all the commits will be squashed into a single commit listing all co-authors.

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

wwylele committed:
feat(Algebra/Order): add Finite(Mul)ArchimedeanClass

YaelDillies commented on Mathlib/Algebra/Order/Archimedean/Class.lean:
```diff
@@ -453,3 +457,112 @@ theorem liftOrderHom_mk (f : M → α) (h : ∀ a b, mk a ≤ mk b → f a ≤ f
 end MulArchimedeanClass
+
+variable (M) in
+/-- Subtype of `MulArchimedeanClass` that removes the class of `1` (i.e. the top element). -/
```
```suggestion
/-- Subtype of `MulArchimedeanClass` that removes the class of `1` (i.e. the top element).

This is useful since the family of non-top archimedean classes is linearly independent.
-/
```
or something like that.

Also, I personally don't like docstrings that say "Same thing as `foo`, but we change bar", because it becomes difficult to understand at once. Actually, now that I reread your docstring for `MulArchimedeanClass`, it suffers the same issue! Could you try making both docstrings more self-contained?

wwylele replied:
Updated the doc

YaelDillies reviewed (commented):
Thanks for splitting!

wwylele committed:
update doc

wwylele committed:
update doc again

wwylele committed:
update doc

YaelDillies reviewed (approved):
Thanks!

maintainer delegate

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Yaël Dillies <yael.dillies@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

wwylele commented:
Small changes since YaelDillies's review on [0d83c45](https://github.com/leanprover-community/mathlib4/pull/28394/commits/0d83c45347b6e613404d8427bcf07337075480fd):
 - Addressed comment on missing space
 - Fixed a "an/a" typo in doc
 - Rebased to latest master to resolve a conflict in the imports

bryangingechen reviewed (approved):
Thanks!
bors d+

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

Co-authored-by: Junyan Xu <junyanxu.math@gmail.com>

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean

wwylele committed:
Update Mathlib/Algebra/Order/Archimedean/Class.lean
